dvc/repo/params/show.py

```diff
@@ -0,0 +1,92 @@
+import copy
+
+from dvc.repo import locked
+from dvc.exceptions import DvcException
+from dvc.dependency.param import DependencyPARAMS
+
+
+class NoParamsError(DvcException):
+    pass
+
+
+def _collect_params(repo, path, recursive, branch):
+    configs = {}
+    for stage in repo.stages:
+        for dep in stage.deps:
+            if not isinstance(dep, DependencyPARAMS):
+                continue
+
+            if dep.path_info not in configs.keys():
+                configs[dep.path_info] = copy.copy(dep)
+                continue
+
+            existing = set(configs[dep.path_info].params)
+            new = set(dep.params)
+            configs[dep.path_info].params = list(existing.update(new))
+    return configs.values()
+
+
+def _read_params(repo, deps, branch):
+    res = {}
+    for dep in deps:
+        assert dep.scheme == "local"
+
+        params = dep._get_info()
+        if not params:
+            continue
+
+        res[str(dep.path_info)] = params
+
+    return res
+
+
+@locked
+def show(
```

Comment:
All of this code is very similar to metrics show, so might look into deduping it.

Reply:
On the other hand there will clearly be some refactoring in https://github.com/iterative/dvc/issues/3572 , so I will just leave it as it is and will unify in the followup PR.